src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.validation.DataValidator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private final DataValidator validator;
+
+    public DataConverterImpl(DataValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        validator.validate(inputData);
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+
+        for (String line : inputData.subList(1, inputData.size())) {
```

Comment:
Assumption that data starts from the second element in 'inputData' list. This could lead to ignoring important data. Assumptions about the data format should be documented clearly or handled dynamically.